src/canvas/Canvas.ts

```diff
@@ -1470,32 +1469,34 @@ export class Canvas extends SelectableCanvas implements CanvasOptions {
             return false;
           }
         }
-        if (target.group === activeSelection) {
+        if (target.group === activeObject) {
           // `target` is part of active selection => remove it
-          activeSelection.remove(target);
+          activeObject.remove(target);
           this._hoveredTarget = target;
           this._hoveredTargets = [...this.targets];
-          if (activeSelection.size() === 1) {
+          if (activeObject.size() === 1) {
             // activate last remaining object
-            this._setActiveObject(activeSelection.item(0) as FabricObject, e);
+            // deselecting the active selection will remove the remaining object from it
+            this._setActiveObject(activeObject.item(0) as FabricObject, e);
           }
         } else {
           //  `target` isn't part of active selection => add it
-          activeSelection.multiSelectAdd(target);
-          this._hoveredTarget = activeSelection;
+          activeObject.multiSelectAdd(target);
+          this._hoveredTarget = activeObject;
           this._hoveredTargets = [...this.targets];
         }
         this._fireSelectionEvents(prevActiveObjects, e);
       } else {
         (activeObject as IText).exitEditing &&
           (activeObject as IText).exitEditing();
         // add the active object and the target to the active selection and set it as the active object
-        activeSelection.multiSelectAdd(activeObject, target);
-        this._hoveredTarget = activeSelection;
+        const newActiveSelection = new ActiveSelection();
```

Comment:
this bothers me. How can we avoid importing the active selection on the canvas?

Reply:
i would use the class registry if it wasn't that then you have to import the activeSelection in order to trigger the registration.

Reply:
No i think that the classRegistry is the correct thing to do here:
- avoid forced imports
- allow for eventual subclassing
I was wrong in saying that you have to import it, the fact that we actually exports it will grant its presence in the classRegistry

Reply:
> I was wrong in saying that you have to import it, the fact that we actually exports it will grant its presence in the classRegistry

This is not precise. We want tree shaking and therefore we want the classRegistry to not account as a side effect so we do want an explicit import to trigger the ActiveSelection import

I agree that we should use the class registry to allow subclassing

Reply:
i meant we don't want to break the splittability, and with current setup ActiveSelection will work anyway

at least we can do a script
for custom builds and the active selection would just work.